pkg/occlient/occlient_test.go

```diff
@@ -4343,3 +4343,172 @@ func TestWaitAndGetDC(t *testing.T) {
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: true,
+			actions: 1,
+		},
+		{
+			name: "Case 3 - Generate and create the BuildConfig but fail with unable to parse image name",
+			args: args{
+				builderImage: "::",
+				namespace:    "testing",
+				gitURL:       "https://github.com/openshift/ruby",
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+					Labels: map[string]string{
+						"app":                              "apptmp",
+						"app.kubernetes.io/component-name": "ruby",
+						"app.kubernetes.io/component-type": "ruby",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				envVars: []corev1.EnvVar{
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: true,
+			actions: 1,
+		},
+		{
+			name: "Case 4 - Generate and create the BuildConfig and pass in no envVars",
+			args: args{
+				builderImage: "ruby:latest",
+				namespace:    "testing",
+				gitURL:       "https://github.com/openshift/ruby",
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+					Labels: map[string]string{
+						"app":                              "apptmp",
+						"app.kubernetes.io/component-name": "ruby",
+						"app.kubernetes.io/component-type": "ruby",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				envVars: []corev1.EnvVar{},
+			},
+			wantErr: false,
+			actions: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientSet := FakeNew()
+
+			fakeClientSet.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(tt.args.commonObjectMeta.Name, tt.args.commonObjectMeta.Namespace, []string{"latest"}), nil
+			})
+
+			// Run function CreateBuildConfig
+			bc, err := fakeClient.CreateBuildConfig(tt.args.commonObjectMeta, tt.args.builderImage, tt.args.gitURL, tt.args.envVars)
+
+			if err == nil && !tt.wantErr {
+				// Check to see how many actions are being ran
+				if (len(fakeClientSet.ImageClientset.Actions()) != tt.actions) && (tt.wantErr != true) {
+					t.Errorf("expected %v action(s) in CreateBuildConfig got %v: %v", tt.actions, len(fakeClientSet.ImageClientset.Actions()), fakeClientSet.ImageClientset.Actions())
+				}
+				// Check that BuildConfig passed back the correct name.
+				if bc.ObjectMeta.Name != tt.args.commonObjectMeta.Name {
```

Comment:
Checks for labels and annotations too would be really nice (for extra surety) :D

Reply:
Okay! :+1: